### ISSUE TITLE
504 Gateway Time-out nginx

### ISSUE THREAD

**Issue by dannyscarborough** [ONLYOFFICE/DocumentServer#433]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
I've just installed the latest version of the document server on my Windows server but I'm receiving this error message when I try to visit http://localhost "504 Gateway Time-out nginx"

How can I troubleshoot this?

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**

Installed all of the perquisites referenced on your instruction page (https://helpcenter.onlyoffice.com/server/windows/document/install-office-apps.aspx)

Setup the PostgreSQL as instructed

Installed  5.3.2

**What is the expected behavior?**

I believe I should see a message to say the document server is running.

**Did this work in previous versions of DocumentServer?**

New installation

**DocumentServer version:**

5.3.2

**Operating System:**

Windows Server 2012 R2

**Browser version:**

IE11 and Chrome 69

**Comment by ShockwaveNN** [comment]
@dannyscarborough Hi, could you zip log folder `C:\Program Files\ONLYOFFICE\DocumentServer\Log` and attach to issue

**Comment by dannyscarborough** [comment]
@ShockwaveNN Thanks for getting back. I'm a little reluctant to upload my logs but I can see this error in the out.log files for converter and docservice every few seconds:

`[2018-11-20 09:36:48.709] [ERROR] nodeJS - [AMQP] Error: connect ECONNREFUSED 127.0.0.1:5672
    at Object._errnoException (util.js:992:11)
    at _exceptionWithHostPort (util.js:1014:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1186:14)
`

I've open the TCP port 5672 on the server and restarted both services but I'm still seeing the error.

**Comment by ShockwaveNN** [comment]
@dannyscarborough 5672 port is rabbitmq port, could you check is it working correctly.
Please note what rabbitmq has some issues on Windows - https://www.rabbitmq.com/windows-quirks.html

**Comment by dannyscarborough** [comment]
> @dannyscarborough 5672 port is rabbitmq port, could you check is it working correctly.
> Please note what rabbitmq has some issues on Windows - https://www.rabbitmq.com/windows-quirks.html

RabbitMQ was failing to start as a service and think this was because of the version of erlang I had installed (19.1) , I've updated to version OTP 21.1 and the service is running but I'm still seeing the error.

I've come across this similar issue on your forum (http://dev.onlyoffice.org/viewtopic.php?f=3&t=11631) but I'm not sure if I'm using an advanced version of erlang.


I've downloaded the latest version from here:

 https://www.erlang.org/downloads

**Comment by dannyscarborough** [comment]
> Please note this table with compatible version between Erlang and RabbitMQ https://www.rabbitmq.com/which-erlang.html

According to that table it should 17.5 😕

**Comment by ShockwaveNN** [comment]
> According to that table it should 17.5 confused

Seems you are right (maybe our error or maybe erlang change thier document), but we checked DocumentServer on windows on those exact version.
We try to update recommended version, but we need time to check everything :)

**Comment by dannyscarborough** [comment]
I think this issue was something specific to our server setup, I've installed it successfully on a 2008 R2 server. Thanks for your help @ShockwaveNN

**Comment by ShockwaveNN** [comment]
@dannyscarborough Glad to hear

**Comment by T0MuX** [comment]
Hi, I want to re-open this issue because I'm habing exactly the same one, and on a Windows Server 2012R2.

Context & versions : 
- Erlang v22 on localhost
- RabbitMQ v3.5.4 on locahost
- Redis v3.2.100 on localhost
- PostgreSQL on another server (a dedicated linux database server on the same network)
- OnlyOffice Document server installed with the following command : `onlyoffice-documentserver.exe /DS_PORT=8020` (because the 80 port is already used on this server)

I took a look into the nginx configuration (ds.conf) and the listening port is 8020 `listen 0.0.0.0:8020;`
Also, in the http-common.conf file, I have this : 
```
upstream docservice {  
  server localhost:8000;
}

upstream spellchecker {  
  server localhost:8080;
}
```

I tried with this instead, but dint change anything :
```
upstream docservice {  
  server localhost:8021;
}

upstream spellchecker {  
  server localhost:8022;
}
```

I also tried with this, in nginx.conf, didnt change anything neither :
`keepalive_timeout 65;`

I don't know at all what is going on. I wait for the page load.. it take a good while (1 or 2 minutes) before having the timeout 504 Gateway Time-out error page.. and I see the URL becoming http://localhost:8020/welcome/ so there is something behind anyway.
